activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -116,6 +116,26 @@ def self.name; "Temp"; end
     ActiveRecord::Base.belongs_to_required_by_default = original_value
   end
 
+  def test_default
+    david = developers(:david)
+    jamis = developers(:jamis)
+
+    model = Class.new(ActiveRecord::Base) do
+      self.table_name = "ships"
+      def self.name; "Temp"; end
+      belongs_to :developer, default: -> { david }
+    end
+
+    ship = model.create!
+    assert_equal david, ship.developer
+
+    ship = model.create!(developer: jamis)
+    assert_equal jamis, ship.developer
+
+    ship.update!(developer: nil)
+    assert_equal david, ship.developer
```

Comment:
Updating to `nil` not mentioned in docs and I think it can lead to confusions.

Let's say I have code where I pass relation by variable to update. Some problem can occur and that variable can be nil. I will expect to get error instead of using default value when I pass explicitly wrong value (`nil`).